### PR TITLE
Transform row-oriented table to columnar-oriented table

### PR DESCRIPTION
**Pull request by chaoyli**
Apache Doris has not supported row-oriented table any more.
In future release, row-oriented table should be transformed to columnar-oriented table.
Now we provide a schema change operator to accomplish this transformation.
Also we provide a script to generate all command about transformation, it should be performed before upgrade.
Script is stored in docs/row_to_column/ directory.

### CODE REVIEW

**Review by imay — CHANGES_REQUESTED** [review]
I have remove binary file from code.

File: tools/row_to_column/get_ddl_stmt.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply tools/row_to_column/get_ddl_stmt.py]: have add license to it

File: tools/row_to_column/pkg_resources.py [hunk 2]
```diff
@@ -0,0 +1,2677 @@
+"""Package resource API
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply tools/row_to_column/pkg_resources.py]: have add license to it.

File: be/src/olap/olap_engine.cpp [hunk 3]
```diff
@@ -266,6 +266,117 @@ OLAPStatus OLAPEngine::load_one_tablet(
     return OLAP_SUCCESS;
 }
 
+void OLAPEngine::check_none_row_oriented_table(const std::vector<OlapStore*>& stores) {
+    std::vector<std::thread> threads;
```
Comment (imay) [inline be/src/olap/olap_engine.cpp]: `threads` is not used

File: be/src/olap/olap_header_manager.cpp [hunk 4]
```diff
@@ -126,19 +126,7 @@ OLAPStatus OlapHeaderManager::set_converted_flag(OlapStore* store) {
...
-        // key format: "hdr_" + tablet_id + "_" + schema_hash
-        split_string<char>(key, '_', &parts);
-        if (parts.size() != 3) {
-            LOG(WARNING) << "invalid header key:" << key << ", splitted size:" << parts.size();
-            return true;
-        }
-        TTabletId tablet_id = std::stol(parts[1].c_str(), NULL, 10);
-        TSchemaHash schema_hash = std::stol(parts[2].c_str(), NULL, 10);
-        return func(tablet_id, schema_hash, value);
-    };
```
Comment (imay) [inline be/src/olap/olap_header_manager.cpp]: You can't change it, this encapsulate `tablet_id` and `schema_hash` encoding logic. If you put it logic outside, every one need to implement the same logic

File: be/src/olap/store.cpp [hunk 5]
```diff
@@ -543,17 +543,74 @@ OLAPStatus OlapStore::_load_table_from_header(OLAPEngine* engine, TTabletId tabl
...
+    auto traverse_header_func = [this, engine](const std::string& key, const std::string& value) -> bool {
+        std::vector<std::string> parts;
+        // key format: "hdr_" + tablet_id + "_" + schema_hash
+        split_string<char>(key, '_', &parts);
+        if (parts.size() != 3) {
+            LOG(WARNING) << "invalid header key:" << key << ", splitted size:" << parts.size();
+            return true;
+        }
+        TTabletId tablet_id = std::stol(parts[1].c_str(), NULL, 10);
+        TSchemaHash schema_hash = std::stol(parts[2].c_str(), NULL, 10);
```
Comment (imay) [inline be/src/olap/store.cpp]: Please use old interface

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/olap_engine.cpp (hunk 3 above)
Comment (chaoyli) [reply be/src/olap/olap_engine.cpp]: I have remove it.

File: be/src/olap/store.cpp (hunk 5 above)
Comment (chaoyli) [reply be/src/olap/store.cpp]: I will reset to last version.

File: be/src/olap/olap_header_manager.cpp (hunk 4 above)
Comment (chaoyli) [reply be/src/olap/olap_header_manager.cpp]: OK, I will reset it to last version.

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**